src/viewProviders/baseModels/multiViewBase.ts

```diff
@@ -0,0 +1,94 @@
+import { TreeItem } from "vscode";
+import { ContextValues, setContextValue } from "../../context/values";
+import { BaseViewProviderData } from "./base";
+import { EnvironmentedBaseViewProviderData, ParentedBaseViewProvider } from "./parentedBase";
+
+export abstract class ViewProviderDelegate<M extends string, T extends BaseViewProviderData> {
+  abstract readonly mode: M;
+
+  abstract readonly viewTitle: string;
+  abstract loadingMessage: string;
+
+  abstract parent: ParentedBaseViewProvider<EnvironmentedBaseViewProviderData, T>;
+  abstract children: T[];
+
+  abstract fetchChildren(element?: T): Promise<T[]>;
+
+  getChildren(element?: T): T[] {
+    if (!this.parent.resource) {
+      return [];
+    }
+    return this.parent.filterChildren(element, this.children);
```

Comment:
Revise this so that ViewProviderDelegate::getChildren() just returns this.children, period, and then have MultiModeViewProvider::getChildren() responsible for making the call to get the current delegate's children, then itself making its own call to its filter.

Less intertwining that way.